modules/spark/src/main/scala/org/apache/spark/sql/ignite/IgniteExternalCatalog.scala

```diff
@@ -271,9 +259,7 @@ private[ignite] class IgniteExternalCatalog(defaultIgniteContext: IgniteContext)
 
     /** @inheritdoc */
     override protected def doCreateTable(tableDefinition: CatalogTable, ignoreIfExists: Boolean): Unit = {
-        val ignite = igniteOrDefault(tableDefinition.identifier.database.getOrElse(DEFAULT_DATABASE), default)
-
-        igniteSQLTable(ignite, tableDefinition.identifier.table) match {
+        igniteSQLTable(ignite, tableDefinition.identifier.table, tableDefinition.identifier.database) match {
```

Comment:
We should check the schema name in `tableDefinition` here.
If the schema name is not `PUBLIC` we should throw.

Reply:
Good point - that's done now, with appropriate tests added.